components/button/button-add.js

```diff
@@ -59,30 +79,131 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			button:${unsafeCSS(getFocusPseudoClass())} .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
+		`;
+	}
 
-			.content {
+	constructor() {
+		super();
+
+		this.hintNearbyDistance = DEFAULT_HINT_DISTANCE;
+		this.hintTiming = HINT_TIMING.ALWAYS;
+		this.textVisible = false;
+
+		this._buttonId = getUniqueId();
+		this._hasFocusOrHover = false;
+		this._visibleDistance = 0;
+
+		this.addEventListener('focus', this._onFocusHover);
+		this.addEventListener('mouseover', this._onFocusHover);
+		this.addEventListener('blur', this._onBlur);
+		this.addEventListener('mouseout', this._onBlur);
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		const buttonSpacing = this._getButtonSpacing();
+		const text = this.text || this.localize('components.button-add.addItem');
+		const content = this.textVisible ? this._renderWithTextVisible(text) : this._renderWithTextHidden(text);
+		const id = !this.textVisible ? this._buttonId : undefined;
+
+		return html`
+			<button
+				class="d2l-label-text d2l-visible-on-ancestor-target"
+				id="${ifDefined(id)}"
+				style="${styleMap(buttonSpacing)}"
+				type="button">
+				<div class="line"></div>
+				${content}
+			</button>
+		`;
+	}
+
+	_getButtonSpacing() {
+		if (this.textVisible || this.hintTiming !== HINT_TIMING.NEARBY) return {};
+
+		this._visibleDistance = this.hintNearbyDistance || DEFAULT_HINT_DISTANCE;
+		return {
+			marginBottom: `-${this._visibleDistance}px`,
+			marginTop: `-${this._visibleDistance}px`,
+			paddingBottom: `${this._visibleDistance}px`,
+			paddingTop: `${this._visibleDistance}px`
+		};
+	}
+
+	_onBlur() {
+		this._hasFocusOrHover = false;
+	}
+	_onFocusHover() {
+		this._hasFocusOrHover = true;
+	}
+
+	_renderWithTextHidden(text) {
+		if (this.textVisible) return this._renderWithTextVisible();
+
+		const visibleOnAncestor = this.hintTiming === HINT_TIMING.NEARBY || this.hintTiming === HINT_TIMING.NEVER;
+		const offset = -(this._visibleDistance - 18);
+
+		return html`
+			<d2l-button-add-icon-text
+				?show-focus="${this._hasFocusOrHover}"
+				?visible-on-ancestor="${visibleOnAncestor}"
+			></d2l-button-add-icon-text>
+			<d2l-tooltip class="vdiff-target" offset="${offset}" for="${this._buttonId}" for-type="label">${text}</d2l-tooltip>
+		`;
+	}
+
+	_renderWithTextVisible(text) {
+		if (!this.textVisible) return this._renderWithTextHidden();
+
+		return html`
+			<d2l-button-add-icon-text
+				text="${text}"
+				?show-focus="${this._hasFocusOrHover}"
+			></d2l-button-add-icon-text>
+		`;
+	}
+
+}
+customElements.define('d2l-button-add', ButtonAdd);
+
+/**
+ * @ignore
+ */
+class ButtonAddIconText extends VisibleOnAncestorMixin(LitElement) {
+	static get properties() {
+		return {
+			text: { type: String },
+			showFocus: { type: Boolean, reflect: true, attribute: 'show-focus' }
+		};
+	}
+
+	static get styles() {
+		return [visibleOnAncestorStyles, css`
+			:host {
 				align-items: center;
 				background-color: white;
```

Comment:
There's no guarantee that the consumer's background color will be white. We might end up needing this to be `transparent` if it looks weird for selected list-items, but I know the reason is to "break" the line. We might end up needing more than one line, for the left & right, unless we can find another clever way to do it.

Reply:
One option would be to figure out the background color at runtime. I would favour two lines if necessary though, because calculating the background color can be costly and imperfect.